doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/DocumentableWrapper.kt

```diff
@@ -0,0 +1,331 @@
+ * writing modified docs to files.
+ * Modify either in immutable fashion using [copy], or in mutable fashion using [asMutable].
+ *
+ * All other properties are read-only and based upon the source-documentable.
+ *
+ * @property documentable The documentable. Should be given.
+ * @property source The source. Should be given.
+ * @property logger Dokka logger that's needed for [findClosestDocComment]. Should be given.
+ *
+ * @property docComment The doc comment found from [documentable] with [findClosestDocComment]
+ * @property path The path of the [documentable].
+ * @property extensionPath If the documentable is an extension function/property:
+ *   "(The path of the receiver).(name of the [documentable])".
+ * @property file The file of the [documentable]'s [source].
+ * @property fileText The text of the [documentable]'s [file].
+ * @property docTextRange The text range of the [file] where the original comment can be found.
+ * @property docIndent The amount of spaces the comment is indented with.
+ *
+ * @property docContent Just the contents of the comment, without the `*`-stuff.
+ * @property tags List of tag names present in this documentable.
+ * @property isModified Whether the [docContent] was modified.
+ *
+ * @see MutableDocumentableWrapper
+ */
+@Suppress("DataClassPrivateConstructor")
+open class DocumentableWrapper internal constructor(
+    val documentable: Documentable,
+    val source: DocumentableSource,
+    private val logger: DokkaConsoleLogger,
+
+    val docComment: DocComment?,
+    val path: String,
+    val extensionPath: String?,
+    val file: File,
+    val fileText: String,
+    val docTextRange: TextRange?,
+    val docIndent: Int?,
+
+    open val docContent: DocContent,
+    open val tags: Set<String>,
+    open val isModified: Boolean,
+) {
+
+    companion object {
+        fun createOrNull(
+            documentable: Documentable,
+            source: DocumentableSource,
+            logger: DokkaConsoleLogger,
+        ): DocumentableWrapper? {
+            val docComment = findClosestDocComment(
+                element = source.psi,
+                logger = logger,
+            )
+
+            val path = documentable.dri.path
+            val extensionPath = documentable.dri.extensionPath
+            val file = File(source.path)
+
+            if (!file.exists()) {
+                // println("Warning: Could not find file for $path")
+                return null
+            }
+
+            val fileText: String = file.readText()
+
+            val docTextRange = docComment?.textRange?.let { ogRange ->
+                // docComment.textRange is the range of the comment in the file, but depending on the language,
+                // it might not exactly match the range of the comment from /** to */. Let's correct that.
+                val query = ogRange.substring(fileText)
+                val startComment = query.indexOf("/**")
```

Comment:
I think Dokka won't detect it then. I just added this indexOf stuff since Dokka is sometimes inconsistent with Kotlin/Java for which part belongs to the Doc and which not (like spaces in front, newline after, etc.)
That said, if it cannot find it, it throws an error for now, since usually something is very broken on the user-side if both dokka and I cannot detect a comment like this.